packages/multiple-did-identities/src/tests/multi-did-identities.spec.ts

```diff
@@ -0,0 +1,123 @@
+// multiple-did-identities.test.ts
+import { DIDManager, DIDIdentity } from '../lib/multiple-did-identities';
+import { generateDidKey } from '../lib/did-key-generator';
+import { encryptData, initializeEncryptionKey } from '../lib/encryption-util';
+import { StorageFactory } from '@adorsys-gis/storage'; // Import StorageFactory
+
+// Mock the necessary modules
+jest.mock('../lib/did-key-generator');
+jest.mock('../lib/encryption-util');
+jest.mock('@adorsys-gis/storage'); //
+
+class MockStorageFactory {
+  insert = jest.fn().mockResolvedValue(undefined);
+  findOne = jest.fn().mockResolvedValue(null);
+  delete = jest.fn().mockResolvedValue(undefined);
+}
```

Comment:
There is a mock solution for the IndexedDB storage that I believe to be more reliable. Check Bolan's `jest.setup.ts` file please.